core/src/main/scala/org/http4s/HeaderV2.scala

```diff
@@ -242,6 +242,8 @@ final class Headers(val headers: List[Header.Raw]) extends AnyVal {
       }
     }
 
+  def foreach(f: Header.Raw => Unit): Unit = headers.foreach(f)
```

Comment:
Are we avoiding dump helpers like this?  I didn't bring back `nonEmpty`.

Reply:
I removed them all, mostly to see how painful it would be. I think adding helpers as we find the need for them is perfectly fine, I just couldn't distinguish between those that were there because they were needed, and those that were there because we wanted to create a collection type